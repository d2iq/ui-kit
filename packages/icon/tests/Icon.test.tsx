import React from "react";
import { shallow } from "enzyme";
import * as emotion from "emotion";
import { createSerializer } from "jest-emotion";
import toJson from "enzyme-to-json";
import { Icon } from "../";
import { SystemIcons } from "../../icons/dist/system-icons-enum";
import { iconSizeL } from "../../design-tokens/build/js/designTokens";

expect.addSnapshotSerializer(createSerializer(emotion));

describe("Icon", () => {
  it("renders", () => {
    const component = shallow(
      <Icon
        shape={SystemIcons.ArrowDown}
        color="blue"
        size={iconSizeL}
        ariaLabel="test icon"
      />
    );

    expect(toJson(component)).toMatchSnapshot();
  });
  it("renders with defaults", () => {
    const component = shallow(<Icon shape={SystemIcons.ArrowDown} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
