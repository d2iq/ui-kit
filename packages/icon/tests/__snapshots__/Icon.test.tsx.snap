// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders 1`] = `
.emotion-0 {
  fill: blue;
}

.emotion-0 use {
  pointer-events: none;
}

<svg
  aria-label="test icon"
  className="emotion-0"
  height={48}
  preserveAspectRatio="xMinYMin meet"
  role="img"
  viewBox="0 0 48 48"
  width={48}
>
  <use
    xlinkHref="#system-arrow-down"
  />
</svg>
`;

exports[`Icon renders with defaults 1`] = `
.emotion-0 {
  fill: #1B2029;
}

.emotion-0 use {
  pointer-events: none;
}

<svg
  aria-label="system-arrow-down icon"
  className="emotion-0"
  height={24}
  preserveAspectRatio="xMinYMin meet"
  role="img"
  viewBox="0 0 24 24"
  width={24}
>
  <use
    xlinkHref="#system-arrow-down"
  />
</svg>
`;
